packages/common/src/ui/layout/tables/components/DataRow/DataRow.tsx

```diff
@@ -1,68 +1,102 @@
+
+  return { isExpanded, toggleExpanded };
+};
+
 export const DataRow = <T extends DomainObject>({
   columns,
   onClick,
   rowData,
   rowKey,
+  ExpandContent,
 }: DataRowProps<T>): JSX.Element => {
   const hasOnClick = !!onClick;
+  const { isExpanded } = useExpanded(rowData.id);
 
   const onRowClick = () => onClick && onClick(rowData);
 
   return (
-    <TableRow
-      sx={{
-        alignItems: 'center',
-        height: '40px',
-        maxHeight: '45px',
-        boxShadow: 'inset 0 0.5px 0 0 rgba(143, 144, 166, 0.5)',
-        padding: '0px 20px',
-        display: 'flex',
-        flex: '1 0 auto',
-      }}
-      onClick={onRowClick}
-      hover={hasOnClick}
-    >
-      {columns.map(column => {
-        return (
-          <TableCell
-            key={`${rowKey}${column.key}`}
-            align={column.align}
-            sx={{
-              borderBottom: 'none',
-              justifyContent: 'flex-end',
-              overflow: 'hidden',
-              textOverflow: 'ellipsis',
-              whiteSpace: 'nowrap',
-              padding: 0,
-              paddingRight: '16px',
-              ...(hasOnClick && { cursor: 'pointer' }),
-              flex: `${column.width} 0 auto`,
-              minWidth: column.minWidth,
-              width: column.width,
-            }}
-          >
-            <column.Cell
-              rowData={rowData}
-              columns={columns}
-              column={column}
-              rowKey={rowKey}
-            />
-          </TableCell>
-        );
-      })}
-    </TableRow>
+    <>
+      <TableRow
+        sx={{
+          alignItems: 'center',
+          height: '40px',
+          maxHeight: '45px',
+          boxShadow: 'inset 0 0.5px 0 0 rgba(143, 144, 166, 0.5)',
+          padding: '0px 20px',
+          display: 'flex',
+          flex: '1 0 auto',
+        }}
+        onClick={onRowClick}
+        hover={hasOnClick}
+      >
+        {columns.map(column => {
+          return (
+            <TableCell
+              key={`${rowKey}${column.key}`}
+              align={column.align}
+              sx={{
+                borderBottom: 'none',
+                justifyContent: 'flex-end',
+                overflow: 'hidden',
+                textOverflow: 'ellipsis',
+                whiteSpace: 'nowrap',
+                padding: 0,
+                paddingRight: '16px',
+                ...(hasOnClick && { cursor: 'pointer' }),
+                flex: `${column.width} 0 auto`,
+                minWidth: column.minWidth,
+                width: column.width,
+              }}
+            >
+              <column.Cell
+                rowData={rowData}
+                columns={columns}
+                column={column}
+                rowKey={rowKey}
+              />
+            </TableCell>
+          );
+        })}
+      </TableRow>
+      <Collapse in={isExpanded}>
+        {ExpandContent ? <ExpandContent /> : null}
+      </Collapse>
+    </>
```

Comment:
I don't really like how this component is here and you need to use the `Expand` column to trigger the state for it. I would have preferred that the expand column itself had the `Expand` UI, but I wasn't sure on the markup for that heh

Reply:
yeah. and yeah - not sure either.